### ISSUE TITLE
Train doesn't show up at "Exit only" stations before destination

### ISSUE THREAD

**Issue by MartinLangbecker** [Traewelling/traewelling#2878]
### Describe the bug

Somewhat related to issue #1010, including discussion of relevancy.

I would like to check in to a train at an "Exit only" station before its final destination, for example EC 56 from Berlin Ostbahnhof to Berlin Hbf:

![IMG_20240821_174712](https://github.com/user-attachments/assets/a6aadf06-d526-4f2d-8f28-a82c6b6f7e68)

However, that train does not show up on Traewelling:

![IMG_20240821_175114](https://github.com/user-attachments/assets/c1662f67-62e0-4618-a435-be79bfafe0db)
![IMG_20240821_175203](https://github.com/user-attachments/assets/cf461b1b-6cad-4dfe-8c6a-524f03c70b5f)

I could probably manually create a journey, but that just doesn't feel right. Also, this only appears to happen if there are only "Exit only" stations left before the final destination of the train.

### Steps to reproduce

1. Search for departures at Berlin Ostbahnhof [BHF].
2. Set date and time to 22.08.2024, 16:55.
3. (Optional) Filter for long distance trains.
4. EC 56 does not show up in the resulting list.

**Comment by marhei** [comment]
This is not a bug, it's intentional. That's why it's called "exit only".

**Comment by MartinLangbecker** [comment]
I get where you're coming from. At the same time, I might have a valid ticket for that section and hop on the train anyways. In that case, I would like to document that journey. From what I understand, this option is used so people won't find tickets for that section in DB Navigator and trains won't have to wait until the official departure time before departing. Please correct me if I'm wrong here.

I haven't checked the codebase yet, but I would assume that this is simply a filter setting that would have to be modified. In issue #1010, this was already fixed for intermediate stops with entry/exit only.

**Comment by d3d9** [comment]
I agree with the idea and have previously checked in at exit only stations by modifying the URL of the trip view (click on the departure at a station where it's a regular departure, then change the station id in the URL to the one where you actually depart).

However, fixing this by displaying these departures in the stationboard to be selectable without the URL modification might be harder than you think, since these trains just don't appear in the departure list. So it might be required to do an additional API request for arrivals, or extend the existing request to also get arrivals if possible, and take just those arrivals that don't terminate at the queried station, to be added to the stationboard additionally to the already displayed departures. This might cause too much effort and performance issues though.

Another solution might be to show the selected departure station at the top of the trip view (the one where you select the exit station). It could be some sort of modifiable drop down offering all stations of the trip except the last one, with the queried station selected by default. While most users would not modify it, it could still be nice for everyone as an additional context / confirmation (and maybe some users would then notice they accidentally selected the wrong departure station, e.g. due to bad GPS). With this, if you wanted to check in to the train with an exit only station, you would have to find it at a regular station and then modify the just presented dropdown to the station where you actually get on the train, even if it's an exit only one.  
Alternatively, the departure station could be made editable after check in, not sure whether there's a specific reason it has not been editable yet (of course editing the destination station is way more commonly useful).

**Comment by MrKrisKrisu** [comment]
> I haven't checked the codebase yet, but I would assume that this is simply a filter setting that would have to be modified.

We don't filter anything out. We request the departures of a station from HAFAS and do not even receive the journeys that cannot be boarded.

**Comment by MrKrisKrisu** [comment]
Since we at Träwelling can't do anything about this and can only use the available data, I would like to close this issue.
